procedures/CWFS_Sensitivity_Matrix_Determination.ipynb

```diff
@@ -0,0 +1,675 @@
+    "    x = hexapod_vals['reportedPosition0'].median()\n",
+    "    y = hexapod_vals['reportedPosition1'].median()\n",
+    "    z = hexapod_vals['reportedPosition2'].median()\n",
+    "    u = hexapod_vals['reportedPosition3'].median()\n",
+    "    v = hexapod_vals['reportedPosition4'].median()\n",
+    "    w = hexapod_vals['reportedPosition5'].median()\n",
+    "\n",
+    "    m1 = np.mean(m1_pressure['pressure'])\n",
+    "    dz = np.nan #round((offset['hexapod_z'][-1] - offset['hexapod_z'][-2])/2.,3)\n",
+    "    print(t1,t2,az, el, rot_pos, x, y, z, u, u, w,dz)\n",
+    "    match_data[i] = (f\"{intra_images[i]}.fits\", f\"{extra_images[i]}.fits\", az, el, rot_pos, x, y, z, u, u, w, m1, dz)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Convert to dataframe\n",
+    "df = pd.DataFrame(match_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#  Save to a csv file for easy access/tracking\n",
+    "filename=\"20200123_match_sens_matrix_metadata.csv\"\n",
+    "df.to_csv(filename)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Now reduce the data for each pair to get the zernikes from fitting\n",
+    "## For details on the analysis itself see tstn-015 and example notebook on running latiss_align_cwfs script\n",
+    "### Can restart the notebook here if required so long as the datafile exists."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Requires new pandas version to use, commenting out but leaving here for when new version is used and \n",
+    "# an exception will be thrown.\n",
+    "# df2 = pd.DataFrame.read_csv(\"20200123_match.csv\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df2 = pd.read_csv(filename)\n",
+    "# Need to add all the zernike terms from the fitting that will be done below\n",
+    "df2['zern_defocus_nm'] = np.zeros(len(df2))\n",
+    "df2['zern_xastig_nm'] = np.zeros(len(df2))\n",
+    "df2['zern_yastig_nm'] = np.zeros(len(df2))\n",
+    "df2['zern_xcoma_nm'] = np.zeros(len(df2))\n",
+    "df2['zern_ycoma_nm'] = np.zeros(len(df2))\n",
+    "df2['zern_xtrefoil_nm'] = np.zeros(len(df2))\n",
+    "df2['zern_ytrefoil_nm'] = np.zeros(len(df2))\n",
+    "df2['zern_sphereical_nm'] = np.zeros(len(df2))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
```

Comment:
Why not import these in the first cell alongside all the other imports?

Reply:
because you can start the notebook from the middle if you've already run the top part.. it's just convenience. There is a note saying this just above the import. I also moved the imports a line up and included pandas.